js/client_exporter/xlsx/xlsx_cell_format_helper.js

```diff
@@ -7,7 +7,7 @@ const xlsxCellFormatHelper = {
         let result = null;
         if(typeUtils.isDefined(sourceObj)) {
             let numberFormatId;
-            if(typeUtils.isDefined(sourceObj.numberFormat) && typeUtils.isNumeric(sourceObj.numberFormat)) {
+            if(typeUtils.isDefined(sourceObj.numberFormat) && typeof sourceObj.numberFormat === 'number') {
```

Comment:
You can use the `isNumber = typedUtils.isNumeric,` for check a number type.

Reply:
The '0000' format string formats Number(42) to '0042' string.
The typedUtils.isNumeric('0000') returns true, '0000' is serialized as numFmtId="0000" so the 'Generic' format (its 'id' is '0') is used and Number(42) is shown as '42' string.

Reply:
Why we need `typeUtils.isDefined(sourceObj.numberFormat)` check if the `typeof sourceObj.numberFormat === 'number'` returns `false` for `null` and `void 0`?

Reply:
IsDefined will be removed in https://github.com/DevExpress/DevExtreme/pull/5724